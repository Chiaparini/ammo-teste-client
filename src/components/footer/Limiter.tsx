import React, { useContext } from 'react'
import { CatalogContext } from '../../contexts/store';
import styled from 'styled-components';
import chevronDown from '../../assets/images/chevron-down.svg'
import { colors, sizes } from '../../assets/style/global';

const Wrapper = styled.div`
  position: relative;
  order: 2;

  @media (min-width: ${sizes.md}) {
    order: 1;
  }
`

const Select = styled.select`
  padding: 8px 26px 8px 16px;
  outline: none;
  border-radius: 5px;
  border-color: ${colors.lightGrey};
  color: ${colors.darkGrey};
  background: white;
  appearance: none;
  background: url(${chevronDown}) no-repeat right;
  background-position-x: 153px;
`

export const Limiter = () => {
  
  // eslint-disable-next-line
  const { state, action }: any = useContext(CatalogContext);
  
  const setLimit = (value: string) => {
    action.setSensitiveParams(1, value);
  }
  
  return (
    <Wrapper>
      <Select name="limiter" id="limiter" onChange={e => setLimit(e.target.value)}>
        <option defaultChecked value="8">8 produtos por p치gina</option>
        <option value="16">16 produtos por p치gina</option>
        <option value="32">32 produtos por p치gina</option>
        <option value="64">64 produtos por p치gina</option>
      </Select>
    </Wrapper>
  )
}
  