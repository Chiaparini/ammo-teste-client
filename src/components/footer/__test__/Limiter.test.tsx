import React from 'react';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Limiter } from '../Limiter';

afterEach(cleanup);

describe('Limiter', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Limiter></Limiter>)
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });
});