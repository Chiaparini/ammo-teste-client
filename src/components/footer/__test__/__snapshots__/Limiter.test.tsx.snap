// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Limiter renders correctly 1`] = `
<div
  className="sc-AykKC lPfmr"
>
  <select
    className="sc-AykKD gEetGY"
    id="limiter"
    name="limiter"
    onChange={[Function]}
  >
    <option
      defaultChecked={true}
      value="8"
    >
      8 produtos por p치gina
    </option>
    <option
      value="16"
    >
      16 produtos por p치gina
    </option>
    <option
      value="32"
    >
      32 produtos por p치gina
    </option>
    <option
      value="64"
    >
      64 produtos por p치gina
    </option>
  </select>
</div>
`;
